tests/test_url.py

```diff
@@ -827,6 +839,40 @@ def test_joinpath(base, to_join, expected):
     assert str(url.joinpath(*to_join)) == expected
 
 
+@pytest.mark.parametrize(
+    "base,to_join,expected",
+    [
+        pytest.param("path", "a", "path/a", id="default_default"),
+        pytest.param("path", "./a", "path/a", id="default_relative"),
+        pytest.param("path/", "a", "path/a", id="empty-segment_default"),
+        pytest.param("path/", "./a", "path/a", id="empty-segment_relative"),
+        pytest.param("path", ".//a", "path//a", id="default_empty-segment"),
+        pytest.param("path/", ".//a", "path//a", id="empty-segment_empty_segment"),
+        pytest.param("path//", "a", "path//a", id="empty-segments_default"),
+        pytest.param("path//", "./a", "path//a", id="empty-segments_relative"),
+        pytest.param("path//", ".//a", "path///a", id="empty-segments_empty-segment"),
+        pytest.param("path", "a/", "path/a/", id="default_trailing-empty-segment"),
+        pytest.param("path", "a//", "path/a//", id="default_trailing-empty-segments"),
+        pytest.param("path", "a//b", "path/a//b", id="default_embedded-empty-segment"),
+    ],
+)
+def test_joinpath_empty_segments(base, to_join, expected):
+    url = URL(f"http://example.com/{base}")
+    assert (
+        str(url.joinpath(to_join))
+        == str(url / to_join)
+        == f"http://example.com/{expected}"
```

Comment:
Do we really want a double equality check here? It's always confusing how these are evaluated.

Reply:
I don't mind how it's formatted, but I asked for the test to check that .joinpath() and the `/` operator are both producing the correct output.

Reply:
This comment wasn't about formatting but chained `==`. They often confuse me as I can never remember that corner case when they behave in unexpected ways.

Reply:
Yes, so it could be formatted as:
```
assert str(url / ...) == expected
assert str(url.joinpath(...)) == expected
```

Reply:
Yep.
```python
assert str(url / ...) == expected and str(url.joinpath(...)) == expected
```
would also work, without chaining.